Aida 530

sharknc opened the pull request:
- Documented all of aifutils.py using reStructuredText docstrings
- Required installation of Sphinx in setup.py 
- Ran Sphinx setup and created all appropriate files needed to generate the python documentation
- Created update_documentation.sh script that can be run to generate the latest python documentation
- Updated README.md to include documentation on how to generate both Javadoc and python documentation
- Updated .gitignore to ignore any documentation that was generated in python of Java. 

**Important**
You will need to install Sphinx in your virtual environment using pip install in order to run the update_documentation.sh script.

jakevb commented:
When no _static folder is in the python/docs/source/ folder, Sphinx gives a warning of 
copying static files... WARNING: html_static_path entry '/AIDA-Interchange-Format/python/docs/source/_static' does not exist.  However, documentation seems to be generated correctly.

jakevb commented:
If you could also update the documentation to the mark_ldc_time method which was recently added to master, that would be helpful.

sharknc commented:
> When no _static folder is in the python/docs/source/ folder, Sphinx gives a warning of
> copying static files... WARNING: html_static_path entry '/AIDA-Interchange-Format/python/docs/source/_static' does not exist. However, documentation seems to be generated correctly.

I have updated the config.py file to no longer require this directories. I have also removed these directories from the repo. The script should now work without them.

sharknc commented:
> When no _static folder is in the python/docs/source/ folder, Sphinx gives a warning of
> copying static files... WARNING: html_static_path entry '/AIDA-Interchange-Format/python/docs/source/_static' does not exist. However, documentation seems to be generated correctly.

sharknc commented:
> If you could also update the documentation to the mark_ldc_time method which was recently added to master, that would be helpful.

I updated the documentation for mark_ldc_time. I also updated ldc_Time to be renamed to ldc_time.

jakevb commented on python/aida_interchange/aifutils.py:
```diff
@@ -522,40 +899,49 @@ def _make_aif_justification(g, doc_id, class_type, system, confidence,
-    :return: A list of confidence assertions describing this object.
-  """
+    :param rdflib.graph.Graph g: The underlying RDF model
+    :param rdflib.term.URIRef confidenced_object: he entity from which to retrieve 
```
Typo: "the" not "he"

sharknc replied:
fixed

jakevb commented on python/aida_interchange/aifutils.py:
```diff
@@ -408,27 +700,41 @@ def mark_informative_justification(g, resource, informative_justification):
-    :param none_of_the_above_prob: if not None, the given confidence will be applied for
-    the "none of the above" option.
-    :return: The mutual exclusion assertion.
+    This is a special case of [markAsMutuallyExclusive] where the alternatives are 
```
Should this be mark_as_mutually_exclusive?

sharknc replied:
yep

sharknc replied:
fixed
